### PR TITLE
feat(titusJobSearch): Support searching with titus jobIds

### PR DESCRIPTION
**Pull request by aravindmd**
- Adding support to search based on titus jobIds.
- The `jobCache` here is populated only when `titus` is enabled 
- The `cache` has a mapping for the `jobId:serverGroupKey`
- `types=jobIds` to be used for this search

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy [hunk 1]
```diff
@@ -50,17 +51,16 @@ class CatsSearchProvider implements SearchProvider, Runnable {
...
-    [:]
-  )
+  private final ConcurrentHashMap<String, Collection<String>> cachedIdentifiersByType = new ConcurrentHashMap<>()
 
   private final FiatPermissionEvaluator permissionEvaluator
   private final List<KeyParser> keyParsers
 
   private final ScheduledExecutorService scheduledExecutorService
 
+  @Value('${titus.enabled:false}')
```
Comment (robzienert) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy]: Do not use `@Value` annotations. Create a `ConfigurationProperties` class and inject that in instead.

Comment (robzienert) [reply clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy]: Also, no. Do not leak cloud providers into core code.

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy [hunk 2]
```diff
@@ -128,13 +130,39 @@ class CatsSearchProvider implements SearchProvider, Runnable {
...
       log.info("Refreshed Cached Identifiers (found ${instanceIdentifiers.size()} instances)")
     } catch (Exception e) {
       log.error("Unable to refresh cached identifiers (instances)", e)
     }
+    //Populate jobId cache only when titus is enabled
+    if (titusEnabled) {
+      try {
+        log.info("Refreshing Cached Identifiers (jobIds)")
+        def jobCache = [:]
+        def instanceIdentifiers = providers.find { provider ->
```
Comment (asher) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy]: I think this should be called `serverGroupIdentifiers`

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy [hunk 3]
```diff
@@ -128,13 +130,39 @@ class CatsSearchProvider implements SearchProvider, Runnable {
...
+          provider.supportsSearch('serverGroups', Collections.emptyMap())
```
Comment (asher) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy]: @robzienert's feedback that cloud provider specific logic should be avoided in core is valid. This specific line shows the way, in that should should be able to provide a `SearchProvider` in `clouddriver-titus` that implements the following logic.

**Comment by aravindmd** [comment]
Going to address some of these comments soon

**Comment by robzienert** [comment]
Ping.

**Comment by aravindmd** [comment]
Closing this , this is prioritized for now , will revisit and address the comments here.

**Pull request closed without merging**